### ISSUE TITLE
How to create an https connection for an http service

### ISSUE THREAD

**Issue by flexbrane** [jwilder/nginx-proxy#687]
Hi,

I have plenty of service on an unique server, so this proxy is really usefull!

I'm hosting a [dillinger](https://hub.docker.com/r/joemccann/dillinger/) instance, who only serve over http. Is it possible to make the proxy establish a secure connection, which is redirected to the 80 port?

Because when I try to add `--env 'VIRTUAL_PROTO=https'` and when I'm trying to go to markdown.mydomain.com, I get `502 Bad Gateway`. 
If I try to go explicitly to `https://markdown.mydomain.com`, I am redirected to another service : blog.mydomain.com which support https.

**Logs when I'm trying to go to markdown.mydomain.com**
```sh
nginx.1    | 2017/01/22 16:15:10 [error] 81#81: *1080 peer closed connection in SSL handshake while SSL handshaking to upstream, client: 1.2.3.4, server: markdown.mydomaincom, request: "GET / HTTP/1.1", upstream: "https://172.17.0.11:8080/", host: "markdown.mydomain.com"
nginx.1    | markdown.mydomain.com 1.2.3.4 - - [22/Jan/2017:16:15:10 +0100] "GET / HTTP/1.1" 502 575 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.67 Safari/537.36"
```

**Docker run**
```sh
docker run -d \
        --name markdown \
        --restart=unless-stopped \
        --volume /etc/localtime:/etc/localtime:ro \
        --env 'VIRTUAL_HOST=markdown.mydomain.com' \
        --env 'LETSENCRYPT_HOST=markedown.mydomain.com' \
        --env 'LETSENCRYPT_EMAIL=xxxxx@mydomain.com' \
        --env 'VIRTUAL_PORT=8080' \
        --env 'VIRTUAL_PROTO=https' \ # Doesn't work (neither http nor https) If I add this line
        joemccann/dillinger:3.13.0
```

**With 'VIRTUAL_PROTO=https', here the generated nginx configuration**
```sh
# markdown.mydomain.com
upstream 57ef69eff869d00b220ed15943efa59757b61394 {
                                ## Can be connect with "bridge" network
                        # markdown
                        server 172.17.0.11:8080;
}
server {
        server_name markdown.mydomain.com;
        listen 80 ;
        access_log /var/log/nginx/access.log vhost;
        location / {
                proxy_pass https://57ef69eff869d00b220ed15943efa59757b61394;
        }
}
```

I understand that nginx try to connect to the dillinger container over https, which is not possible, so a `502 Bad Gateway` error is generated.

**Nginx conf without the line 'VIRTUAL_PROTO=https'**
```sh
# markdown.mydomain.com
upstream 57ef69eff869d00b220ed15943efa59757b61394 {
                                ## Can be connect with "bridge" network
                        # markdown
                        server 172.17.0.11:8080;
}
server {
        server_name markdown.mydomain.com;
        listen 80 ;
        access_log /var/log/nginx/access.log vhost;
        location / {
                proxy_pass http://57ef69eff869d00b220ed15943efa59757b61394;
        }
}
```
Without `'VIRTUAL_PROTO=https'` I can access to the dillinger service, but without a secure connection. Is there a way to create an https connection between a client and the reverse proxy ?

Thank you :-)

**Comment by thomasleveil** [comment]
you need to:
- create a key and server certificate for the `markdown.mydomain.com` domain
- name those files `markdown.mydomain.com.key` and `markdown.mydomain.com.crt`
- mount those files with a docker volumes to `/etc/nginx/certs/`

as described in the [readme](https://github.com/jwilder/nginx-proxy#ssl-support) files.

**Comment by flexbrane** [comment]
I'm using [JrCs/docker-letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion), is there any way to do this automatically, like for the other services?

**Comment by thomasleveil** [comment]
I don't know, I'm not familiar with JrCs/docker-letsencrypt-nginx-proxy-companion. That's a question to ask to the maintainers of that project.

**Comment by flexbrane** [comment]
I will. Thank you!

**Comment by flexbrane** [comment]
Problem solved.

Indeed, it was an [JrCs/docker-letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion) issue. 

I saw the letsencrypt container logs and see that there were an error. The letsencrypt agent couldn't put a temporary file (to validate the subdomain) in the dillinger container.

So I created a temporary very simple web service, as my dillinger subdomain and this time the letsencrypt could put his temporary file. So my certificates where created.

Now I have my secure connection with my dillinger service :)



**Temporary web server to generate the certs**
```sh
docker run -d \
        --name=genCertsMarkdown \
        --volume /home/docker/markdown:/www \
        --env 'VIRTUAL_HOST=markdown.mydomain.com' \
        --env 'LETSENCRYPT_HOST=markdown.mydomain.com' \
        --env 'LETSENCRYPT_EMAIL=xxxxxmydomain.com' \
        --env 'VIRTUAL_PORT=80' \
        androw/uhttpd
```
